src/StreamrClient.js

```diff
@@ -110,10 +134,49 @@ export default class StreamrClient extends EventEmitter {
                 this.options.auth, this.signer, this.getUserInfo()
                     .catch((err) => this.emit('error', err)),
                 (streamId) => this.getStream(streamId)
-                    .catch((err) => this.emit('error', err)), this.options.publisherGroupKeys,
+                    .catch((err) => this.emit('error', err)), this.keyStorageUtil,
             )
         }
 
+        if (this.options.auth.privateKey || this.options.auth.provider) {
+            // subscribing to own inbox stream
+            this.getPublisherId().then((ethAddress) => this.subscribe(ethAddress, async (parsedContent, streamMessage) => {
+                try {
+                    if (streamMessage.contentType === StreamMessage.CONTENT_TYPES.GROUP_KEY_REQUEST) {
+                        if (this.keyExchangeUtil) {
+                            await this.keyExchangeUtil.handleGroupKeyRequest(streamMessage)
+                        }
+                    } else if (streamMessage.contentType === StreamMessage.CONTENT_TYPES.GROUP_KEY_RESPONSE_SIMPLE) {
+                        if (this.keyExchangeUtil) {
+                            const { streamId } = streamMessage.getParsedContent()
+                            // A valid publisher of the client's inbox stream could send key responses for other streams to which
+                            // the publisher doesn't have write permissions. Thus the following additional check is necessary.
+                            // TODO: fix this hack in other PR (and move logic to keyExchangeUtil)
+                            const valid = await this.subscribedStreamPartitions[streamId + '0'].isValidPublisher(streamMessage.getPublisherId())
+                            if (valid) {
+                                await this.keyExchangeUtil.handleGroupKeyResponse(streamMessage)
+                            } else {
+                                throw new InvalidGroupKeyResponseError(
+                                    `Received group key from an invalid publisher ${streamMessage.getPublisherId()} for stream ${streamId}`
+                                )
+                            }
+                        }
+                    } else if (streamMessage.contentType === StreamMessage.CONTENT_TYPES.ERROR_MSG) {
+                        debug('WARN: Received error of type %s from %s: %s',
+                            streamMessage.getParsedContent().code, streamMessage.getPublisherId(), streamMessage.getParsedContent().message)
+                    } else {
+                        throw new InvalidContentTypeError(`Cannot handle message with content type: ${streamMessage.contentType}`)
+                    }
+                } catch (err) {
```

Comment:
This catches a lot, might want to verify the type of the error and rethrow non-matching.